### PR TITLE
Feature: Asynchronous DOM Updates

### PR DESCRIPTION
**Pull request by jaypeasee**
# Pull Request Template

#### What’s this PR do?
This PR introduces asynchronous functionality to reflect new bookings and cancellations on the page without having to reload. Before when i'd make a booking, the change would be reflected by the `DOM` but when I ran another date or name filter, that booking returned. It only went away on reload. This PR fixes that.

#### Where should the reviewer start?
* Review the `bookNewRoom` function on line 424 of `index.js` and see how that invokes the `postNewBooking` method on the `apiCalls` object on line 23 of `apiCalls.js`. From there, look at how the display and fetch functions are called when that `Promise` is resolved.
* Review the `handleBookingCancellation` function on line 455 of `index.js` and see how that invokes the `deleteBooking` method on the `apiCalls` object on line 35 of `apiCalls.js`. From there look at how the display and fetch functions are called when that `Promise` is resolved.

#### How should this be manually tested?
1. In the manager view, pick a date, and book some rooms under a guest name. Run the date filter back to that same date to confirm that room is still no long showing up.
1. In the guest view, pick a date, and book some rooms. Run the date filter back to that same date to confirm that room is still no long showing up.
1. In the manager view, look up a guest name in the search bar (a guest who has future bookings already). Cancel those bookings. Search that same name again to confirm those bookings still no longer show up.

#### Any background context you want to provide?
This PR, I believe finalizes all of my API call functionality and asynchronous code portion of the project.

#### What are the relevant tickets?
* #39 
* #46

### CODE REVIEW

File: src/apiCalls.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ export let apiCalls = {
   },
 
   deleteBooking(bookingToDelete) {
-    fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings', {
+    return fetch('https://fe-apps.herokuapp.com/api/v1/overlook/1904/bookings/bookings', {
```
Comment (jdxsmith) [inline src/apiCalls.js:36]: Good catch on adding the return statement when using a DELETE with the fetch API.

File: src/index.js [hunk 2]
```diff
@@ -416,14 +416,19 @@ function handleRoomBooking(event, nameEntered, guestProfile) {
   if (guestProfile === "error") {
     displayNavFormError(event, "guest name");
   } else {
-    const bookingFormat = createBookingObject(roomName.slice(5), guestProfile.id, currentHotel.date);
-    const newBooking = apiCalls.postNewBooking(bookingFormat);
-    getApiData();
-    displaySuccessfulBooking(event, roomName, currentHotel.date);
+    bookNewRoom(roomName, guestProfile, event)
```
Comment (jdxsmith) [inline src/index.js:419]: Nice use of a helper function to clean up the flow of this function

File: src/index.js [hunk 3]
```diff
@@ -432,21 +437,29 @@ function createBookingObject(roomNumber, guestID, datePicked) {
...
+  const cancelButton = event.target;
   const bookingToCancel = cancelButton.parentNode;
-  const cancellationObject = { id: parseInt(bookingToCancel.dataset.id) }
-  apiCalls.deleteBooking(cancellationObject);
-  displayCancellation(cancelButton)
+  const cancellationObject = { id: parseInt(bookingToCancel.dataset.id) };
+  const newCancellation = apiCalls.deleteBooking(cancellationObject);
+  newCancellation
+    .then(() => displayCancellation(cancelButton))
+    .then(() => updateBookingData(newCancellation));
```
Comment (jdxsmith) [inline src/index.js:462]: Nice job making the deletions asynchronous

**Review by jdxsmith — APPROVED** [review]
Great job making all of the calls asynchronous, your application looks great now that it doesn't refresh upon new API calls.  In `index.js`, there may be room for improvement regarding the placement of your functions, but this is a minor change; just something worth giving a second look.  Your site looks good and functions very smoothly.  Nice work!

**Comment by jaypeasee** [comment]
Thanks @jdxsmith !!

**Pull request merged**